Core/src/org/sleuthkit/autopsy/healthmonitor/ServicesHealthMonitor.java

```diff
@@ -0,0 +1,767 @@
+            createDbInfoTable.append("id SERIAL PRIMARY KEY NOT NULL,");
+            createDbInfoTable.append("name text NOT NULL,");
+            createDbInfoTable.append("value text NOT NULL");
+            createDbInfoTable.append(")");
+            statement.execute(createDbInfoTable.toString());
+            
+            statement.execute("INSERT INTO db_info (name, value) VALUES ('SCHEMA_VERSION', '" + CURRENT_DB_SCHEMA_VERSION.getMajor() + "')");
+            statement.execute("INSERT INTO db_info (name, value) VALUES ('SCHEMA_MINOR_VERSION', '" + CURRENT_DB_SCHEMA_VERSION.getMinor() + "')");
+            
+            conn.commit();
+        } catch (SQLException ex) {
+            try {
+                conn.rollback();
+            } catch (SQLException ex2) {
+                logger.log(Level.SEVERE, "Rollback error");
+            }
+            throw new HealthMonitorException("Error initializing database", ex);
+        } finally {
+            try {
+                conn.close();
+            } catch (SQLException ex) {
+                logger.log(Level.SEVERE, "Error closing connection.", ex);
+            }
+        }
+    }
+    
+    /**
+     * The task called by the ScheduledThreadPoolExecutor to handle
+     * the database writes.
+     */
+    static final class DatabaseWriteTask implements Runnable {
+
+        /**
+         * Write current metric data to the database
+         */
+        @Override
+        public void run() {
+            try {
+                getInstance().writeCurrentStateToDatabase();
+            } catch (HealthMonitorException ex) {
+                logger.log(Level.SEVERE, "Error writing current metrics to database", ex); //NON-NLS
+            }
+        }
+    }
+    
+    /**
+     * Get an exclusive lock for the health monitor database.
+     * Acquire this before creating, initializing, or updating the database schema.
+     * @return The lock
+     * @throws HealthMonitorException 
+     */
+    private CoordinationService.Lock getExclusiveDbLock() throws HealthMonitorException{
+        try {
+            String databaseNodeName = DATABASE_NAME;
+            CoordinationService.Lock lock = CoordinationService.getInstance().tryGetExclusiveLock(CoordinationService.CategoryNode.HEALTH_MONITOR, databaseNodeName, 5, TimeUnit.MINUTES);
+
+            if(lock != null){
+                return lock;
+            }
+            throw new HealthMonitorException("Error acquiring database lock");
+        } catch (InterruptedException | CoordinationService.CoordinationServiceException ex){
+            throw new HealthMonitorException("Error acquiring database lock");
+        }
+    } 
+    
+    /**
+     * Get an shared lock for the health monitor database.
+     * Acquire this before database reads or writes.
+     * @return The lock
+     * @throws HealthMonitorException 
+     */
+    private CoordinationService.Lock getSharedDbLock() throws HealthMonitorException{
```

Comment:
See comments above. Not sure why this is needed. Seems like database level locking might be sufficient.